pkg/templates/translator/terraform_translator.go

```diff
@@ -152,13 +145,35 @@ func (t TerraformTranslator) SchemaOfOriginalType(tp any, opts Options) *openapi
 				o.TypeExpress = te[i]
 			}
 
-			refs[i] = t.SchemaOfOriginalType(tt, o).NewRef()
+			refs[i] = t.SchemaOfOriginalType(tt, o)
 		}
 
-		s.WithLength(int64(len(ts))).
-			WithItems(&openapi3.Schema{
-				OneOf: refs,
-			})
+		switch {
+		case len(refs) == 1:
+			s.WithItems(refs[0]).
+				WithLength(1)
+		case len(refs) > 1:
+			var diffSchema bool
+
+			for i := 0; i < len(refs); i++ {
+				for j := i + 1; j < len(refs); j++ {
+					if refs[i] != refs[j] {
+						diffSchema = true
+						break
+					}
+				}
+			}
+
+			if diffSchema {
+				s.WithItems(openapi3.NewObjectSchema()).
+					WithLength(int64(len(ts)))
+			} else {
+				s.WithItems(refs[0]).
+					WithLength(int64(len(ts)))
+			}
+		default:
+			s.WithItems(openapi3.NewObjectSchema())
+		}
 
 		s.Title = title
```

Comment:
It seem there common parts in every switch
```
		s.Title = title
		s.Description = description
		s.WriteOnly = sensitive
```

can we set them outside the switch structure put them at the end of the function?